### ISSUE TITLE
Downloaded files view

### ISSUE THREAD

**Issue by dstillman** [zotero/zotero-ios#261]
We've had a few variations on this request:

> Is it possible to add a filter that enable us to only see downloaded PDF files? Say “Downloaded files” or “On My Device” is helpful to me.

https://github.com/zotero/zotero-ios/issues/220 will be an option for some people, but for people who want to stick to download-as-needed mode, something like like would make sense.

I don't think a top-level virtual collection would be ideal, though — it seems like you'd want to be able to see everything within a given collection with a file downloaded, in addition to the whole library. So one option would be a filter option like iOS Mail has, where you can turn it on and then choose among various filter options. For us that could include downloaded files, "favorites"/"unread" (if we add first-class (i.e., non-colored-tag) support for those), and maybe some other things.

Another approach would be to have an area appear below the search bar when focused that would let you choose some non-text-based options — the options above, and maybe a [tag selector](https://github.com/zotero/zotero-ios/issues/39) would be accessible from there as well.

Basically we need to come up with a more comprehensive plan for searching/filtering in general. This is a fairly tricky design problem, so we'll probably want to mock up various options.

**Comment by michalrentka** [comment]
@dstillman I guess it's time to implement nicer popover settings for items (and later PDF as well). PSPDFKit has a nice popover for their settings:
![IMG_457E7A51340E-1](https://user-images.githubusercontent.com/1022956/114833482-ed4d7680-9dcf-11eb-9bf5-22b3f1bfaf63.jpeg)

We could have:
"Sort By - Title  >"
"Sort Order - Ascending | Descending"
"Downloads Only - "switch""

The first one would push the picker screen with sort options. 
Second one would ideally have icons, but since our design abilities are now limited we could do with iOS default:
![](https://miro.medium.com/max/2000/1*lE2BhFZpBP0w4AVBXk-MKw.png)
And third one would shown just a switch to toggle between downloads only.

Anyway when this setting is enabled, do we want to show items with main attachment downloaded or any attachment downloaded? If we go with any downloaded, that may cause confusion in case you don't have the main attachment downloaded and it shows up there.

**Comment by dstillman** [comment]
The problem with just doing this in a single popover combined with sort options is that it really needs to show when it's in a filtered state, like Unread in Mail. Relatedly, when we add a tag selector, how are we going to show that a tag is selected?

Seems like filter options need to be in a separate UI element from the sort options.

> do we want to show items with main attachment downloaded or any attachment downloaded?

That's tricky. Neither seems ideal. It's bad to find out you don't have a main attachment you thought you had, but it's also bad if you're trying to clear up space and something doesn't show up as downloaded. Former seems worse, but if the main attachment isn't downloaded, that's at least indicated in the icon, so maybe it makes more sense to show for any?

**Comment by michalrentka** [comment]
> Former seems worse, but if the main attachment isn't downloaded, that's at least indicated in the icon, so maybe it makes more sense to show for any?

I think that makes is even worse and people will report as a bug. They'll set filtering for downloads only and see the icon indicating that the attachment is not downloaded. 

If they want this feature for space cleaning purposes we could improve the settings screen and add the option to delete individual item files there.

**Comment by dstillman** [comment]
OK, that's fair. So we can do main file only for now.

**Comment by michalrentka** [comment]
So I tried implementing a simple filtering popup similar to mail app:
![Simulator Screen Shot - iPad Pro (11-inch) (2nd generation) - 2021-04-16 at 16 30 58](https://user-images.githubusercontent.com/1022956/115040043-50730200-9ed1-11eb-8504-34759d74eb26.png)

**Comment by stakats** [comment]
Wouldn’t it make more sense to put the filter icon up top with the sort icon?

**Comment by michalrentka** [comment]
There's not really that much space on top in portrait on smaller devices. I'd argue we could move "Sort" and "Add" icons to bottom. They are easily accessible there and we'd have more space for search and "Select"/"Done" on top. 

Or we could go the default way of having search bar underneath the navigation bar (as it is currently on iPhone) and we could have more icons on top. But then we'd have different sizes for navigation bars in left and right part of the app. The search bar can be hidden and would show on drag, but then it's not really discoverable, even though I'd say people are used to this and would probably figure it out.

**Comment by dstillman** [comment]
> Or we could go the default way of having search bar underneath the navigation bar (as it is currently on iPhone) and we could have more icons on top.

We couldn't do this on an iPhone 12 mini, at least. We're pretty much out of room already.

Maybe we don't have a choice here but to have a bottom bar like Mail. I might keep Sort at the top and just have Filter and Add at the bottom, the same as Mail, to keep it symmetrical. We could even show sync status in the middle in a similar way as Mail.

**Comment by michalrentka** [comment]
@dstillman we already show sync-related stuff on the left side in the toolbar. We could show something like "Last Synced: Date", but I'm not sure that's needed. We could also move sync-related stuff here, but this screen is not visible when item detail is opened, while collections/libraries are visible. So moving the "+" button on bottom is probably best.

**Comment by dstillman** [comment]
Right, I was suggesting moving it. As for not being visible when in item detail, that's true, though Mail does it, and I'm not sure you particularly need to see sync status when you're busy editing an item. The main reason to move it, I think, would be to avoid repeatedly popping up a toolbar and hiding it, which is arguably more distracting than just updating the status in the already existing status bar.

"Last Synced: [date]" — again, similar to Mail, and maybe sets the mind at ease. But I agree that it doesn't seem all that necessary as long as we're showing sync errors.

It could also be a place to say how many items are currently displayed after any filters. Not that important, but it's something you can't see now, because the collections-list count only shows all items, not filtered items.

> So moving the "+" button on bottom is probably best.

I was saying Filter and Add would be at the bottom, and Sort would be at the top. Did you mean Sort?

**Comment by michalrentka** [comment]
> As for not being visible when in item detail, that's true, though Mail does it, and I'm not sure you particularly need to see sync status when you're busy editing an item.

It let's you see that the changes you made are actually synced. Although people probably don't watch that too much.

> It could also be a place to say how many items are currently displayed after any filters. Not that important, but it's something you can't see now, because the collections-list count only shows all items, not filtered items.

This could work.

>> So moving the "+" button on bottom is probably best.

> I was saying Filter and Add would be at the bottom, and Sort would be at the top. Did you mean Sort?

I meant it would be best to move the add button from top to bottom (and keep sort on top) as you suggested.

**Comment by michalrentka** [comment]
@dstillman so how important it is to actually see the sync progress? I added it to items view, but now the editing mode is interfering with sync progress, because you have to see edit actions. When editing is off, the filter will interfere with sync progress. Only when you're pretty much doing nothing you'll see the sync progress.

I think it would be better to keep sync progress in toolbar on the left side of the screen. But we should keep the left toolbar visible all the time. If sync is not in progress then we should keep some static message like Apple Mail does: "Last Sync: 5 minutes ago". This way you'll always see sync going on. You'll see filter/editing related stuff on right side. And the best thing imo is that both toolbars will be visible and it'll be nicer visually. But this will break when you open item detail. In this case we can either hide the toolbar on left side too or just keep it there.

**Comment by michalrentka** [comment]
These probably look better than only toolbar on the right side

![Simulator Screen Shot - iPad Pro (11-inch) (2nd generation) - 2021-04-20 at 15 17 51](https://user-images.githubusercontent.com/1022956/115402800-0bf6a780-a1ec-11eb-84c3-2039f75a1ab8.png)
![Simulator Screen Shot - iPad Pro (11-inch) (2nd generation) - 2021-04-20 at 15 17 48](https://user-images.githubusercontent.com/1022956/115402804-0ef19800-a1ec-11eb-9613-c516f489ce4a.png)

**Comment by dstillman** [comment]
I dunno — that's a whole lot of chrome. I think it's too heavy.

What does it look like with just the right-hand side?

**Comment by michalrentka** [comment]
@dstillman looks worse to me

![Simulator Screen Shot - iPad Pro (11-inch) (2nd generation) - 2021-04-21 at 10 14 06](https://user-images.githubusercontent.com/1022956/115520338-57f82980-a28a-11eb-84a0-c241c9ffe377.png)

**Comment by michalrentka** [comment]
@flachware do you have any suggestions on these? Do you find any of above to be ok?

**Comment by flachware** [comment]
When you run out of space at the top the next step is to add a toolbar at the bottom; I think the latest version could work. But I find it inconsistent to have the plus icon down there while it is on top in the master view – maybe swap it with the sort icon? Then filter and sort would be at the bottom.

The toolbar at the bottom does not have to be so heavy. I’d make it less high so the icons are centered vertically. Shrink the font size a lot (also in the activity pane in the master view), like in Apple Mail. Try removing the border and perhaps use even a white background for the toolbar.

**Comment by michalrentka** [comment]
@flachware Apple Mail does have smaller fonts, icons are the same size, so I don't recommend making the whole bar smaller. iOS 12 actually increased the size from 44 to 50 from previous version. Icons are centered vertically, but there is extra space for the slider on bottom, that's default iOS behavior.

I like the white color, it makes it a bit easier on eyes:

![Simulator Screen Shot - iPad Pro (11-inch) (3rd generation) - 2021-04-30 at 15 11 21](https://user-images.githubusercontent.com/1022956/116699850-60590e80-a9c6-11eb-8659-462ea2dfb9db.png)

I personally prefer, when both toolbars are visible, but here is a screenshot without the left one:
![Simulator Screen Shot - iPad Pro (11-inch) (3rd generation) - 2021-04-30 at 15 17 09](https://user-images.githubusercontent.com/1022956/116700528-29cfc380-a9c7-11eb-859c-85b5ba8c6e64.png)

@dstillman fyi

**Comment by michalrentka** [comment]
And with sync on right side with and without left panel

![1](https://user-images.githubusercontent.com/1022956/116701873-b4fd8900-a9c8-11eb-8f63-ae036c37a1bd.jpg)
![2](https://user-images.githubusercontent.com/1022956/116701884-b7f87980-a9c8-11eb-87e1-98cfd93e3cd7.jpg)

**Comment by flachware** [comment]
@michalrentka ah I don’t have that slider, I didn’t know that. The smaller font size looks good to me. I’d go with a single toolbar at the right side with white background (the toolbar in the mockup still has a very light shading of 1%). I think the sync status has to stay at the left side and the left toolbar should be hidden unless there is something to report.

**Comment by michalrentka** [comment]
@dstillman do you agree with the above? I think that could work.

**Comment by dstillman** [comment]
I like the smaller font. Showing sync status on left and not showing when nothing to show seems OK to me. Not totally sure about pure-white — seems a little weird not to distinguish it from the list views at all, but we can try it.

Would we use the middle space in the right-hand bar for conveying anything?

I think I agree with @flachware on consistent placement for the + button, with Sort in the bottom-right instead.

**Comment by michalrentka** [comment]
@dstillman if we're keeping sync on left then I don't know what to put in the middle of the right bar. I guess we could show number of filtered items if filter is on. Otherwise it would be empty.

**Comment by michalrentka** [comment]
@dstillman do we want to go ahead with this version?

**Comment by dstillman** [comment]
Generally, yes, but for what it's worth, we're doing the opposite of Mail: using gray for the top toolbar and (almost) white for the bottom one.

Should we just confirm that we don't like it more the other way around?

**Comment by michalrentka** [comment]
@dstillman it's difficult, on first sight we're doing it the other way around, but both apps are just using default values (except we colored the bottom toolbar to white).
The default navigation bar color is the gray one. Mail app uses the big navigation title, which makes it look different. If you just scroll up in the mail app, you'll see the same default navigation bar that we have. In this case the only difference is that we used white color for the toolbar.
